pypit/core/arskysub.py

```diff
@@ -132,6 +132,143 @@ def bg_subtraction_slit(slit, slitpix, edge_mask, sciframe, varframe, tilts,
+
+    # Sky pixels for fitting
+    fit_sky = (thismask == True) & (ivar > 0.0) & (inmask == True)
+    isrt = np.argsort(piximg[fit_sky])
+    wsky = piximg[fit_sky][isrt]
+    sky = image[fit_sky][isrt]
+    sky_ivar = ivar[fit_sky][isrt]
+
+    # Restrict fit to positive pixels only and mask out large outliers via a pre-fit to the log
+    if (POS_MASK==True):
+        pos_sky = (sky > 1.0) & (sky_ivar > 0.0)
+        if np.sum(pos_sky) > nspec:
+            lsky = np.log(sky[pos_sky])
+            lsky_ivar = lsky * 0. + 0.1
+
+            # Init bspline to get the sky breakpoints (kludgy)
+            tmp = pydl.bspline(wsky[pos_sky], nord=4, bkspace=bsp)
+
+            #skybkpt = bspline_bkpts(wsky[pos_sky], nord=4, bkspace=bsp $
+            #, / silent)
+            if False:
+                plt.clf()
+                ax = plt.gca()
+                ax.scatter(wsky[pos_sky], lsky)
+                #ax.scatter(wsky[~full_out], sky[~full_out], color='red')
+                #ax.plot(wsky, yfit, color='green')
+                plt.show()
+                #debugger.set_trace()
+            lskyset, outmask, lsky_fit, red_chi = arutils.bspline_profile(
+                wsky[pos_sky], lsky, lsky_ivar, np.ones_like(lsky),
+                fullbkpt = tmp.breakpoints, upper=sigrej, lower=sigrej,
+                kwargs_reject={'groupbadpix':True})
+            res = (sky[pos_sky] - np.exp(lsky_fit)) * np.sqrt(sky_ivar[pos_sky])
+            lmask = (res < 5.0) & (res > -4.0)
+            sky_ivar[pos_sky] = sky_ivar[pos_sky] * lmask
+
+
+    # Full fit now
+    full_bspline = pydl.bspline(wsky, nord=4, bkspace=bsp)
+    skyset, outmask, yfit, _ = arutils.bspline_profile(wsky, sky, sky_ivar, np.ones_like(sky),
+                                                       fullbkpt=full_bspline.breakpoints,upper=sigrej, lower=sigrej,
+                                                       kwargs_reject={'groupbadpix':True, 'maxrej': 10})
+    if (np.sum(outmask == False) == 19233):
+        import pickle
+        import sys
+        picklefile = '/Users/joe/python/PYPIT-development-suite/dev_algorithms/nb/iterfit_vs_profile.pkl'
+        out_tuple = (image, ivar, thismask, piximg, inmask, bsp, sigrej, wsky, sky, sky_ivar)
+        fileObject = open(picklefile, 'wb')
+        pickle.dump(out_tuple, fileObject)
+        fileObject.close()
+        sys.exit(-1)
+
+    # Evaluate and save
+    bgframe[thismask], _ = skyset.value(piximg[thismask])
+
+    # JFH testing
+#    skyset, outmask, yfit, _ = arutils.bspline_profile(wsky, sky, sky_ivar, np.ones_like(sky),
+#                                                       upper=sigrej, lower=sigrej, maxiter = 20,
+#                                                       kwargs_bspline={'bkspace':bsp},
+#                                                       kwargs_reject={'maxrej': 10})
+
+    #  fullbkpt = full_bspline.breakpoints,
+    from pypit.core.pydl import iterfit as bspline_iterfit
```

Comment:
Please put all the imports at the top of the module.  And don't use 'as' unless it's absolutely necessary.  I don't think it is in this case.

Reply:
Hey guys this is some development stuff in a routine that is not used. I'll fix it in the next version, but this routine is not called at all (global_sksybu). I didn't want to put the import at the top of the module, because this code will disappear when I'm done developing.

I realize this should not be in the master branch, but at the moment we have a bunch of other deprecated garbage in the master branch, and so I don't think this is a big deal.